### PR TITLE
[Communication] - Chat - Merge feature/chat-preview4 into master

### PR DESCRIPTION
**Pull request by glorialimicrosoft**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs [hunk 1]
```diff
@@ -13,9 +13,9 @@ public partial class AddChatParticipantsResult
...
         protected ChatClient() { }
-        public ChatClient(System.Uri endpointUrl, Azure.Communication.CommunicationTokenCredential communicationTokenCredential, Azure.Communication.Chat.ChatClientOptions? options = null) { }
-        public virtual Azure.Response<Azure.Communication.Chat.CreateChatThreadResult> CreateChatThread(string topic, System.Collections.Generic.IEnumerable<Azure.Communication.Chat.ChatParticipant> participants, string? repeatabilityRequestId = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.Communication.Chat.CreateChatThreadResult>> CreateChatThreadAsync(string topic, System.Collections.Generic.IEnumerable<Azure.Communication.Chat.ChatParticipant> participants, string? repeatabilityRequestId = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public ChatClient(System.Uri endpointUrl, Azure.Communication.CommunicationTokenCredential communicationTokenCredential, Azure.Communication.Chat.ChatClientOptions options = null) { }
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs:16]: `endpointUrl` should be called `endpoint`

File: sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs [hunk 2]
```diff
@@ -0,0 +1,198 @@
+namespace Azure.MixedReality.RemoteRendering
```
Comment (RezaJooyandeh) [inline sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs:1]: Why is this file modified?

File: sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs (hunk 1 above)
Comment (juancamilor) [reply sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs:16]: this one is on the list for preview5, no hurt in doing now if we want.

**Review by juancamilor — APPROVED** [review]

File: sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs (hunk 2 above)
Comment (glorialimicrosoft) [reply sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs:1]: I think it was generated after running the api generation script, I will remove it.

File: sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs (hunk 1 above)
Comment (glorialimicrosoft) [reply sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs:16]: perhaps let's do the changes asked from the azure review together?

File: sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs (hunk 2 above)
Comment (glorialimicrosoft) [reply sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs]: removed

File: sdk/communication/Azure.Communication.Chat/tests/Infrastructure/ChatLiveTestBase.cs [hunk 3]
```diff
@@ -9,7 +9,7 @@ namespace Azure.Communication.Chat.Tests
 {
     public class ChatLiveTestBase : RecordedTestBase<ChatTestEnvironment>
     {
-        public ChatLiveTestBase(bool isAsync) : base(isAsync)
+        public ChatLiveTestBase(bool isAsync) : base(isAsync,RecordedTestMode.Playback)
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Chat/tests/Infrastructure/ChatLiveTestBase.cs:12]: we should not hardcode to playback or it cannot run in live mode

File: sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs (hunk 2 above)
Comment (RezaJooyandeh) [reply sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs]: I see that you reverted the changes in a separate commit. It is better to use `git reset HEAD~` till the commit that was making the change, undo the change there and then commit without it. So it does not cause future conflicts for the MixedReality team. Essentially we want to make sure this file is not touched by our branch

File: sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs (hunk 1 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs:16]: Is not it better to just do it instead of having the same comment on another PR? + and one less thing to worry in the future 😉

**Review by jsquire — CHANGES_REQUESTED** [review]
Would you please rebase to remove the Smoke Tests, Event Hubs, and Service Bus areas?  They should not be included as part of this PR.

File: sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs (hunk 2 above)
Comment (juancamilor) [reply sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs]: maybe easier to just get the latest from master, and paste the Chat folder with your changes on top in a new branch, so we avoid any merging issues @glorialimicrosoft ?

File: sdk/communication/Azure.Communication.Chat/tests/Infrastructure/ChatLiveTestBase.cs (hunk 3 above)
Comment (glorialimicrosoft) [reply sdk/communication/Azure.Communication.Chat/tests/Infrastructure/ChatLiveTestBase.cs]: removed it, was mistakenly committed before

File: sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs (hunk 2 above)
Comment (glorialimicrosoft) [reply sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs]: @juancamilor, that might work expect that Reza also had a commit in the preview4 branch that is outside the Chat folder. I will see what I can do to get rid of the merging issues.

Comment (RezaJooyandeh) [reply sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs]: The problem seems to have originated from the master branched being merged to your branch early one. You should avoid merging base branch to a child branch and instead rebase your branch onto master.

Comment (glorialimicrosoft) [reply sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs]: Ya... I think that was the problem that caused the merging issues. Thanks, Reza, I will take out that merge and do rebase instead.

Comment (glorialimicrosoft) [reply sdk/mixedreality/Azure.MixedReality.RemoteRendering/api/Azure.MixedReality.RemoteRendering.netstandard2.0.cs]: Actually, I think that would require some force pushes... I will do it properly in a separate branch then create a brand new PR

**Comment by glorialimicrosoft** [comment]
Created a clean PR that resolves the merging issues in current branch, please review the new PR instead: https://github.com/Azure/azure-sdk-for-net/pull/19288. Thank you! @juancamilor @RezaJooyandeh
